src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -572,14 +572,17 @@ export function getInputType(
     function getDiscriminatorValue(
         m: Model,
         baseModel?: InputModelType
-    ): string | undefined {
+    ): string | number | undefined {
```

Comment:
I suggest we add a comment here because this looks a bit weird.
But I assume it makes sense because when the kind is `EnumMember` there is a possibility to be number?

Reply:
Good catch. `number` is not allowed as discriminator property type.